addons/binding/org.openhab.binding.hpprinter/src/main/java/org/openhab/binding/hpprinter/internal/handler/HttpsTrustManager.java

```diff
@@ -0,0 +1,85 @@
+/**
+ * Copyright (c) 2010-2019 Contributors to the openHAB project
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.openhab.binding.hpprinter.internal.handler;
+
+import java.security.KeyManagementException;
+import java.security.NoSuchAlgorithmException;
+import java.security.SecureRandom;
+import java.security.cert.X509Certificate;
+
+import javax.net.ssl.HostnameVerifier;
+import javax.net.ssl.HttpsURLConnection;
+import javax.net.ssl.SSLContext;
+import javax.net.ssl.SSLSession;
+import javax.net.ssl.TrustManager;
+import javax.net.ssl.X509TrustManager;
+
+/**
+ * The {@link HttpsTrustManager} is responsible for ignoring invalid HTTPS certificates.
+ *
+ * @author Stewart Cossey - Initial contribution
+ */
+
+public class HttpsTrustManager implements X509TrustManager {
+
+    private static TrustManager[] trustManagers;
+    private static final X509Certificate[] ACCEPTED_ISSUERS = new X509Certificate[] {};
+
+    @Override
+    public void checkClientTrusted(X509Certificate[] x509Certificates, String s)
+            throws java.security.cert.CertificateException {
+    }
+
+    @Override
+    public void checkServerTrusted(X509Certificate[] x509Certificates, String s)
+            throws java.security.cert.CertificateException {
+    }
+
+    public boolean isClientTrusted(X509Certificate[] chain) {
+        return true;
+    }
+
+    public boolean isServerTrusted(X509Certificate[] chain) {
+        return true;
+    }
+
+    @Override
+    public X509Certificate[] getAcceptedIssuers() {
+        return ACCEPTED_ISSUERS;
+    }
+
+    public static void allowAllSSL() {
+        HttpsURLConnection.setDefaultHostnameVerifier(new HostnameVerifier() {
```

Comment:
This is not acceptable this will compromise the whole openHAB instance.

Feel free to disable certificate checking for your binding but leave alone the security mechanisme of the system as a whole.

Please look into the ExtensibleTrustManager as used in the icloud binding.